server/jetstream_api.go

```diff
@@ -2156,6 +2196,262 @@ func (s *Server) jsLeaderServerRemoveRequest(sub *subscription, c *client, _ *Ac
 	s.sendAPIErrResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp))
 }
 
+/*
+// Request to have the metaleader provide which streams a server contains
+func (s *Server) jsLeaderServerStreamInfoRequest(sub *subscription, c *client, _ *Account, subject, reply string, rmsg []byte) {
+	if c == nil || !s.JetStreamEnabled() {
+		return
+	}
+
+	ci, acc, _, msg, err := s.getRequestInfo(c, rmsg)
+	if err != nil {
+		s.Warnf(badAPIRequestT, msg)
+		return
+	}
+
+	js, cc := s.getJetStreamCluster()
+	if js == nil && (cc != nil && cc.meta == nil) {
+		return
+	}
+
+	// Extra checks here but only leader is listening.
+	js.mu.RLock()
+	isLeader := cc.isLeader()
+	js.mu.RUnlock()
+
+	if !isLeader {
+		return
+	}
+
+	var resp = JSApiMetaServerStreamInfoResponse{ApiResponse: ApiResponse{Type: JSApiMetaServerStreamInfoType}}
+
+	if isEmptyRequest(msg) {
+		resp.Error = NewJSBadRequestError()
+		s.sendAPIErrResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp))
+		return
+	}
+
+	var req JSApiMetaServerStreamInfoRequest
+	if err := json.Unmarshal(msg, &req); err != nil {
+		resp.Error = NewJSInvalidJSONError()
+		s.sendAPIErrResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp))
+		return
+	}
+
+	if cc == nil {
+		// single server logic
+		if req.Server != s.info.Name {
+			resp.Error = NewJSClusterServerNotMemberError()
+			s.sendAPIErrResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp))
+			return
+		}
+
+		resp.Content = make(map[string][]StreamConfig)
+		resp.Success = true
+		js.mu.RLock()
+		for aName, jsa := range js.accounts {
+			jsa.mu.RLock()
+			for _, s := range jsa.streams {
+				resp.Content[aName] = append(resp.Content[aName], s.cfg)
+			}
+			jsa.mu.RUnlock()
+		}
+		js.mu.RUnlock()
+		return
+	}
+	// clustered code
+
+	var found string
+	js.mu.RLock()
+	for _, p := range cc.meta.Peers() {
+		si, ok := s.nodeToInfo.Load(p.ID)
+		if ok && si.(nodeInfo).name == req.Server {
+			found = p.ID
+			break
+		}
+	}
+	js.mu.RUnlock()
+
+	if found == _EMPTY_ {
+		resp.Error = NewJSClusterServerNotMemberError()
+		s.sendAPIErrResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp))
+		return
+	}
+
+	resp.Content = make(map[string][]StreamConfig)
+	js.mu.Lock()
+	for aName, a := range cc.streams {
+		for _, s := range a {
+			contains := false
+			for _, p := range s.Group.Peers {
+				if p == found {
+					contains = true
+					break
+				}
+			}
+			if contains {
+				resp.Content[aName] = append(resp.Content[aName], *s.Config)
```

Comment:
Why not just inline at line 2288?

Reply:
vs bool contains, etc.

Reply:
same as above, will incorporate suggestions when this is keept.